authentication/src/main/kotlin/com/uber/sdk2/auth/internal/AuthProvider.kt

```diff
@@ -38,63 +42,76 @@ class AuthProvider(
-          body?.takeIf { response.isSuccessful }
-            ?: throw AuthException.ServerError("Bad response ${response.code()}")
-        } ?: PARResponse("", "")
-
-      val queryParams =
-        mapOf(
-          "request_uri" to parResponse.requestUri,
-          "code_challenge" to codeVerifierGenerator.generateCodeChallenge(verifier),
-        )
-
+      val parResponse = sendPushedAuthorizationRequest(ssoConfig)
+      val queryParams = getQueryParams(parResponse)
       val authCode = ssoLink.execute(queryParams)
       when (authContext.authType) {
         AuthType.AuthCode -> AuthResult.Success(UberToken(authCode = authCode))
-        is AuthType.PKCE -> {
-          val tokenResponse =
-            authService.token(
-              ssoConfig.clientId,
-              verifier,
-              authContext.authType.grantType,
-              ssoConfig.redirectUri,
-              authCode,
-            )
-
-          if (tokenResponse.isSuccessful) {
-            tokenResponse.body()?.let { AuthResult.Success(it) }
-              ?: AuthResult.Error(
-                AuthException.ClientError("Token request failed with empty response")
-              )
-          } else {
-            AuthResult.Error(
-              AuthException.ClientError("Token request failed with code: ${tokenResponse.code()}")
-            )
-          }
-        }
+        is AuthType.PKCE -> performPkce(ssoConfig, authContext.authType, authCode)
       }
     } catch (e: AuthException) {
       AuthResult.Error(e)
     }
   }
 
+  private suspend fun performPkce(
+    ssoConfig: SsoConfig,
+    authType: AuthType.PKCE,
+    authCode: String,
+  ): AuthResult {
+    val tokenResponse =
+      authService.token(
+        ssoConfig.clientId,
+        verifier,
+        authType.grantType,
+        ssoConfig.redirectUri,
+        authCode,
+      )
+
+    return if (tokenResponse.isSuccessful) {
+      tokenResponse.body()?.let { AuthResult.Success(it) }
+        ?: AuthResult.Error(AuthException.ClientError("Token request failed with empty response"))
+    } else {
+      AuthResult.Error(
+        AuthException.ClientError("Token request failed with code: ${tokenResponse.code()}")
+      )
+    }
+  }
+
+  private suspend fun sendPushedAuthorizationRequest(ssoConfig: SsoConfig) =
+    authContext.prefillInfo?.let {
+      val response =
+        authService.loginParRequest(
+          ssoConfig.clientId,
+          RESPONSE_TYPE,
+          Base64Util.encodePrefillInfoToString(it),
+          ssoConfig.scope ?: "profile",
+        )
+      val body = response.body()
+      body?.takeIf { response.isSuccessful }
+        ?: throw AuthException.ServerError("Bad response ${response.code()}")
+    } ?: PARResponse("", "")
+
+  private fun getQueryParams(parResponse: PARResponse) = buildMap {
+    when (authContext.authType) {
+      AuthType.AuthCode -> {
+        parResponse.requestUri.takeIf { it.isNotEmpty() }?.let { put(REQUEST_URI, it) }
```

Comment:
Seems this is the case for both, so you could take it out of the `when` clause as well

Reply:
yea, i realized it. 
there's another common query parameter that i am working on so will change in the next PR.